cmd/skaffold/app/cmd/deploy.go

```diff
@@ -30,11 +33,33 @@ func NewCmdDeploy(out io.Writer) *cobra.Command {
 		Args:  cobra.NoArgs,
 		RunE: func(cmd *cobra.Command, args []string) error {
 			// Same actions as `skaffold run`, but with pre-built images.
-			return run(out)
+			opts.Command = "deploy"
+			return runDeploy(out)
 		},
 	}
 	AddRunDevFlags(cmd)
 	AddRunDeployFlags(cmd)
 	cmd.Flags().StringSliceVar(&opts.PreBuiltImages, "images", nil, "A list of pre-built images to deploy")
 	return cmd
 }
+
+func runDeploy(out io.Writer) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
```

Comment:
in run.go, we also have a `defer plugin.CleanupClients()` here. Should it be added?

Reply:
no it should not be added here since the plugins should not be called in deploy flow, But i can check again.

Reply:
I’m in the process of removing the plugin code, so you don’t need to add it!